client/app/components/parameters.html

```diff
@@ -19,7 +19,11 @@
           <option ng-repeat="option in extractEnumOptions(param.enumOptions)" value="{{option}}">{{option}}</option>
         </select>
       </span>
+      <span ng-switch-when="query" ng-controller="QueryBasedParameterController">
+        <select ng-model="param.value" class="form-control" ng-options="queryResult for queryResult in queryResults">
+        </select>
+      </span>
       <input ng-switch-default type="{{param.type}}" class="form-control" ng-model="param.ngModel">
     </span>
-  </div>
+  </
```

Comment:
Good catch. Not sure how to catch this in dev. It did not break stuff when I deployed it locally. Are there some good tricks? I realize that frontend does not have good test coverage, but are there tools that I can use?